### PR TITLE
Adding TypeScript to existing project.

### PR DESCRIPTION
**Pull request by SolomiiaLomnytska22**
**Refs:**
#3 (Integrate TypeScript in your project.)
**Proposed changes:**
Using TypeScript in the project instead of JavaScript. Updating all the components using TypeScript syntax.

Tested locally. All existing functionality is supported.

https://github.com/SolomiiaLomnytska22/solomiia-training/assets/136481621/964e9fe6-2623-459a-92d3-a20fdf4538b1

### CODE REVIEW

File: app/src/Users.vue [hunk 1]
```diff
@@ -17,13 +17,14 @@
...
 
-<script>
+<script lang="ts">
 import axios from 'axios'
 import AddUserModal from './components/users/AddUserModal.vue'
 import UserTable from './components/users/UserTable.vue'
 import Button from './components/common/Button.vue'
 import Component from 'vue-class-component'
 import Vue from 'vue'
+import UserModel from "@/models/UserModel";
```
Comment (marina-grechukha) [inline app/src/Users.vue]:
```suggestion
import UserModel from '@/models/UserModel'
```

Comment (SolomiiaLomnytska22) [reply app/src/Users.vue]: Thanks, just fixed it in the most recent commit!

File: app/package.json [hunk 2]
```diff
@@ -37,10 +42,11 @@
...
-      "eslint:recommended"
+      "eslint:recommended",
+      "@vue/typescript"
     ],
     "parserOptions": {
-      "parser": "@babel/eslint-parser"
+      "parser": "@typescript-eslint/parser"
     },
     "rules": {
       "vue/multi-word-component-names": "off"
```
Comment (kaytee-te) [inline app/package.json]: I would suggest adding these five rules for better formatting.

`"space-before-function-paren": 2,
 "object-curly-spacing": ["error", "always"],
 "array-bracket-spacing": ["error", "always"],
 "computed-property-spacing": ["error", "always"],
"quotes": ["error", "single"]`

File: app/src/components/header/Header.vue [hunk 3]
```diff
@@ -42,18 +42,18 @@ import Vue from 'vue'
...
 export default class Header extends Vue {
-  showMenu = false
-  userInfo = {
+  showMenu: boolean = false
+  userInfo: { name: string; avatar: string } = {
     name: 'John Doe',
     avatar: 'Man.jpg'
   }
-  menuItems = [
+  menuItems: { id: number; route: string; title: string }[] = [
```
Comment (kaytee-te) [inline app/src/components/header/Header.vue]: Maybe it makes sense to add an interface for the `menuItems`.

File: app/package.json (hunk 2 above)
Comment (SolomiiaLomnytska22) [reply app/package.json]: Thanks, added those rules in recent commit!

File: app/src/components/header/Header.vue (hunk 3 above)
Comment (SolomiiaLomnytska22) [reply app/src/components/header/Header.vue]: Thanks, added this interface in the new commit.

**Comment by kaytee-te** [comment]
lgtm!

**Comment by marina-grechukha** [comment]
LGTM, just one suggestion - how about keeping all types and interfaces in one file `app/src/types.ts` instead of creating a separate one for each?

**Review by kaytee-te — APPROVED** [review]

**Review by marina-grechukha — APPROVED** [review]

**Pull request merged**